src/components/VariantSelector.svelte

```diff
@@ -12,7 +13,19 @@
 
   const change = () => {
     selectedVariant = variants.find((v) => v.id === selectedId);
-
+    $pageState = {
+      ...$pageState,
+      // because both app channel and ping use `id` as the identifier
+      // we check if selectedVariant.channel exists to differentiate
+      // between setting channel and ping state
```

Comment:
This is a little confusing. I feel like maybe we should handle this in the parent? Perhaps via some helper functions to update the store? We could probably use the dispatch function to get notified when this changes.